frontend/src/metabase/visualizations/visualizations/SmartScalar.jsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import { Box, Flex } from "grid-styled";
+import { t, jt } from "c-3po";
+import _ from "underscore";
+
+import { formatNumber, formatValue } from "metabase/lib/formatting";
+import colors from "metabase/lib/colors";
+
+import Icon from "metabase/components/Icon";
+
+import * as Query from "metabase/lib/query/query";
+import * as Card from "metabase/meta/Card";
+import { parseFieldBucketing, formatBucketing } from "metabase/lib/query_time";
+
+import { columnSettings } from "metabase/visualizations/lib/settings/column";
+import {
+  MinRowsError,
+  NoBreakoutError,
+} from "metabase/visualizations/lib/errors";
+
+import ScalarValue, {
+  ScalarWrapper,
+  ScalarTitle,
+} from "metabase/visualizations/components/ScalarValue";
+
+export default class Smart extends React.Component {
+  static uiName = "Smart Scalar";
+  static identifier = "smartscalar";
+  static iconName = "smartscalar";
+
+  static minSize = { width: 3, height: 3 };
+
+  static noHeader = true;
+
+  _scalar: ?HTMLElement;
+
+  static settings = {
+    ...columnSettings({
+      getColumns: ([{ data: { cols } }], settings) => [
+        _.find(cols, col => col.name === settings["scalar.field"]) || cols[1],
+      ],
+    }),
+    "scalar.switch_positive_negative": {
+      title: t`Switch positive / negative colors?`,
+      widget: "toggle",
+    },
+  };
+
+  // Always return true here so that we don't discourage this feature.
+  static isSensible() {
+    return true;
+  }
+
+  // Smart scalars need to have a breakout
+  static checkRenderable(series, settings) {
```

Comment:
I think this should have the same sort of `insights` check.

```
if (!series[0].data.insights) {
  // throw something
}}
```

I'm not sure the rest of the checks are even necessary.